src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -31,6 +39,10 @@ internal sealed partial class ServiceHubRemoteHostClient : RemoteHostClient
         private readonly JsonRpc _rpc;
         private readonly ReferenceCountedDisposable<RemotableDataJsonRpc> _remotableDataRpc;
 
+        private readonly object _globalNotificationsGate = new object();
+        private Task<GlobalNotificationState> _globalNotificationsTask = Task.FromResult(GlobalNotificationState.NotStarted);
```

Comment:
Might be good to add a comment explaining what this serialization is attempting to address.